lib/ipaddr.rb

```diff
@@ -352,6 +408,11 @@ def hash
     return ([@addr, @mask_addr].hash << 1) | (ipv4? ? 0 : 1)
   end
 
+  # Returns the network mask of the IPAddr as an IPAddr::Netmask object
+  def mask_addr
+    return Netmask.new(@mask_addr, @family)
```

Comment:
Maybe, that would probably depend on the situations in which it is called. Keep in mind that caching also means every location that could change it have to destroy the cache (in this case it looks to be limited to `IPAddr#mask!` I don't think the readability of the lib improves by caching.